app/models/user.rb

```diff
@@ -0,0 +1,32 @@
+require 'bcrypt'
+
+class User
+
+  include DataMapper::Resource
+
+  property :id, Serial
+  property :user_name, String, unique: true,
+                               message: 'This Username is already taken'
+  property :email, String, unique: true,
+                           message: 'This Email is already taken'
+  property :password_digest, Text
+  property :password_token, Text
+  property :password_token_timestamp, Time
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  validates_confirmation_of :password, message: "Sorry, your password does not
```

Comment:
Prefer the new style validations `validates :column, confirmation: value` over `validates_confirmation_of`.